File: Teledyne DALSA/Sapera/Demos/Classes/Vc/ColorConversionDemo/SapMyProcessing.cpp
//	SapMyProcessing.cpp : implementation file
//

#include "SapMyProcessing.h"


//
// Constructor/Destructor
//
SapMyProcessing::SapMyProcessing(SapBuffer *pBuffers, SapColorConversion* pColorConv, SapProCallback pCallback, void *pContext)
   : SapProcessing(pBuffers, pCallback, pContext)
{
   m_ColorConv = pColorConv;
}

SapMyProcessing::~SapMyProcessing()
{
   if (m_bInitOK) 
      Destroy();
}


//
// Processing Control
//
BOOL SapMyProcessing::Run()
{
   if (m_ColorConv->IsSoftwareEnabled())
   {
      m_ColorConv->Convert( GetIndex());
   }

   return TRUE;
}

File: Teledyne DALSA/Sapera/Demos/NET/ColorConversionDemo/CSharp/SapMyProcessing.cs
using System;
using System.Collections.Generic;
using System.Text;
using DALSA.SaperaLT.SapClassBasic;

namespace DALSA.SaperaLT.Demos.NET.CSharp.ColorConvDemo
{
   class SapMyProcessing : SapProcessing   
   {
      private SapColorConversion m_ColorConv;
     // Constructor

      public SapMyProcessing(SapBuffer pBuffers, SapColorConversion pColorConv, SapProcessingDoneHandler pCallback, Object pContext)
        : base(pBuffers)
     {
        base.ProcessingDoneEnable = true;
        base.ProcessingDone += pCallback;
        base.ProcessingDoneContext = pContext;
        m_ColorConv = pColorConv;
     }

     public override bool Run()
     {
        if (m_ColorConv != null && m_ColorConv.Initialized && m_ColorConv.SoftwareEnabled)
        {
           m_ColorConv.Convert(base.Index);
        }

        return true;
     }
 
   }
}
